src/Player/Player.cpp

```diff
@@ -37,7 +40,145 @@ namespace Player {
+                    print_hand.emplace_back("Card " + to_string(i) + " : Type = " + this->hand->get_cards().at(i)->getType_to_string() + ", Country: " + this->hand->get_cards().at(i)->getCountry() + ".");
+                }
+
+                //Prompt the user to choose three cards
+                answer_cards.emplace_back() = Terminal::print_select("Please input your desired [3] cards: "); /* Verify if user can actually input 3 values */
+
+                //Checking if cards are valid
+               /* if(this->hand->cardsValidForExchange(*//* IS VALID FOR EXCHANGE *//*)){
+                    exchange_desired = Terminal::print_select("Player [Color] has cards that are valid for exchange, would you like to proceed now?");
+                    if(exchange_desired){
+                        army_exchange += this->hand->exchange(*//*  INDICES *//* );
+                    }
+                }*/
+
+                //Exchanging cards
+                while(this->hand->size() >= 5){
+                    army_exchange += this->hand->exchange(&answer_cards[0]); /* ??????? check if conversion is correct */
+                }
+            }
+
+        //Therefore, the total armies available (A) to Player are
+        army_total = army_country + army_continent + army_exchange;
+
+        /*
+         * Then player places all armies on some of the countries it owns as they see fit.
+         */
+
+        //Looping around the number of armies available to place
+        do{
+            //Display available options (armies available + countries for positioning)
+            Terminal::print("You are now able to place your reinforcement armies. The total armies and countries available for positioning are the following: ");
+                vector<string> armies_print;
+                armies_print.emplace_back(army_total + " total armies available");
+                Terminal::print(armies_print);
+            /*INDEX = */
+
+                //Gathering countries owned by player and in map
+                    //vector<Countries *> countries_owned ~ already declared above
+                    //vector<Countries *> countries_in_map ~ already declared above
+                //Printing Utility
+                vector<string> options;
+                for(auto c : countries_owned) {
+                    options.emplace_back(c->get_name() + " (which has" + to_string(c->get_armies()) + " armies present).");
+                }
+                //Asking player to input data
+                options.emplace_back("\nPlease enter the country [#] to which you would like to place armies: ");
+                int answer_country = Terminal::print_select(options);
+
+                vector<string> options_number;
+                options_number.emplace_back("Enter how many armies you would like to place on " + countries_in_map.at(placement_answer)->get_name());
+                int answer_number = Terminal::print_select(options_number);
```

Comment:
Here you should limit the range of valid answers as the user should not be able to enter a number greater than the number of armies owned.
Check print_select for overload with minimum and maximum values defined to limit the input range.

Reply:
As aforementioned, for the range of valid answers I don't have the most recent Terminal in my branch. Should i manually update it??

Reply:
As per your suggestion I merged develop into my branch. The code now looks like this:
>                 int answer_number = Terminal::print_select(1, army_total, "Enter how many armies you would like to place on your chosen country: ");